### DISCUSSION TITLE
How to get cursor postion from a MultiLineEntry ?

### DISCUSSION THREAD

**Discussion by imjakey** [fyne-io/fyne#4362] [Q&A] [answered]
Hi every one,

I am developing an instant messaging application, and in order to implement the "@" mention feature in group chats, I need to display a dropdown list of people at the cursor position when the user types the "@" character in the input box. I have searched through all the documentation, but I couldn't find an API to get the current cursor position in the input box. How can I implement this functionality?

thank's a lot!

like this 
![image](https://github.com/fyne-io/fyne/assets/2662273/e8f18a77-4cab-48e2-ac01-74ecd06c8972)

**Comment by Jacalz** [comment]
You can use .CursorRow and .CursorColumn on your entry object.
![image](https://github.com/fyne-io/fyne/assets/25466657/45ca25bc-3cc2-4cec-930b-1ada373384ad)
https://pkg.go.dev/fyne.io/fyne/v2/widget#Entry

**Comment by imjakey** [comment]
@Jacalz  Thank you for replay, in my case i need to show the popup window at the cursor position, how can i convert entry.CursorRow and  entry.CursorColumn to a fyne.Position or any other method to get the absolute position of the cursor? 

i have tried this piece of code, but not work:

`pos := fyne.NewPos(float32(entry.CursorRow), float32(entry.CursorColumn))`
`popup.ShowAtPosition(pos)`


where did I go wrong?

**Reply by dweymouth** [reply]
The CursorRow and Column are just the positions in terms of text characters. You'd need to do a lot of math to convert those to on-screen position (i.e. fyne.Position). I'd have to think about if there's a way to do that with the public APIs

**Reply by Jacalz** [reply]
You might be able to use the `fyne.MeasureText` call to find out how far in you are. If you use the cursor row for finding out vertical offset, you can measure the text up to the cursor to know where you are horizontally. I can't quite remember how to get the absolute position of the widget right now though. Adding that together should get you close I think.

**Reply by imjakey** [reply]
Thanks again ! I'm really confused how to do this, please consider adding a public API .

**Reply by dweymouth** [reply]
I think it would get even more complicated than @jacalz suggestion with scrolling. I think it's a reasonable feature request to have a `CursorPosition() fyne.Position` API - please open a feature request ticket @imjakey with the same context from this discussion

**Reply by andydotxyz** [reply]
Scrolling is a good point. Would the position be relative to the visible corner or the hidden source corner?
For the purpose of displaying a popup neither are the full answer as the absolute position is required. From the driver you can get absolute for the containing widget, but not including the scroll offset...

**Reply by imjakey** [reply, marked as answer]
@dweymouth  I create it at  #4382 

Thanks for all the help.
